Pegasus/include/Math.hpp

```diff
@@ -30,13 +659,450 @@ class JacobiEigenvalue
+         * @return HEDS face iterator
+         */
+        HalfEdgeDataStructure::face_iterator GetHedsFaceIterator() const
+        {
+            return m_hedsFaceIterator;
+        }
+
+        /**
+         * @brief HyperPlane getter.
+         * @return hyperplane
+         */
+        HyperPlane const& GetHyperPlane() const
+        {
+            return m_hyperPlane;
+        }
+
+        /**
+         * @brief Face indices getter.
+         * @return face indices array
+         */
+        std::array<size_t, 3> const& GetIndices() const
+        {
+            return m_indices;
+        }
+
+    private:
+        std::vector<typename Vertices::iterator> m_vertices;
+        typename Vertices::iterator m_extremalVertex;
+        size_t m_extremalVertexIndex;
+        HalfEdgeDataStructure::face_iterator m_hedsFaceIterator;
+        HyperPlane m_hyperPlane;
+        std::array<size_t, 3> m_indices;
+    };
+
+    /**
+     * @brief Constructs a quickhull convex hull calculation algorithm object.
+     * @param vertices vertex buffer object
+     */
+    explicit QuickhullConvexHull(Vertices& vertices)
+        : m_vertices(vertices)
+    {
+    }
+
+    /**
+     * @brief Calculates a convex hull from a given vertex buffer.
+     */
+    void Calculate()
+    {
+        CalculateInitialGuess();
+        Refine();
+    }
+
+    /**
+     * @brief Returns convex hull faces.
+     * @return convex hull faces.
+     */
+    Faces const& GetFaces() const
+    {
+        return m_faces;
+    }
+
+    /**
+     * @brief Returns convex hull vertices.
+     * @return convex hull vertices.
+     */
+    std::list<typename Vertices::iterator> const& GetVertices() const
+    {
+        return m_convexHullVertices;
+    }
+
+private:
+    Faces m_faces;
+    Vertices& m_vertices;
+    std::list<typename Vertices::iterator> m_convexHullVertices;
+    glm::dvec3 m_mean;
+    HalfEdgeDataStructure m_heds;
+    std::unordered_map<
+        HalfEdgeDataStructure::Face*, typename Faces::iterator
+    > m_hedsFaceIteratorMap;
+
+    /**
+     * @brief Calculates initial guess tetrahedron and performs initalization.
+     */
+    void CalculateInitialGuess()
+    {
+        //Calculate covariance matrix and it's eigenvectors
+        m_mean = CalculateExpectedValue(m_vertices.begin(), m_vertices.end());
+        JacobiEigenvalue const eigenvalue(CalculateCovarianceMatrix(m_vertices.begin(), m_vertices.end(), m_mean));
+        glm::dmat3 eigenvectorsNormalized = eigenvalue.GetEigenvectors();
+        eigenvectorsNormalized = {
+            glm::normalize(eigenvectorsNormalized[0]),
+            glm::normalize(eigenvectorsNormalized[1]),
+            glm::normalize(eigenvectorsNormalized[2]),
+        };
+
+        //Calculate extremal vertices
+        std::array<typename Vertices::iterator, 3> maximaVertices, minimaVertices;
+        FindExtremalVertices(m_vertices.begin(), m_vertices.end(), eigenvectorsNormalized, maximaVertices, minimaVertices);
```

Comment:
Method's signature is

```c++
void FindExtremalVertices(
    Iterator begin, Iterator end,
    glm::dmat3 const& basis,
    std::array<Iterator, 3>& minimaVertices,
    std::array<Iterator, 3>& maximaVertices)
```

But you supply `maximaVertices, minimaVertices`. Is this intended?